app/src/main/java/com/example/android/oscarine/signup/SignupFragment.kt

```diff
@@ -36,10 +36,8 @@ class SignupFragment : Fragment() {
             navigateToSigninFragment()
         }
 
-        viewModel.response.observe(this, Observer {
-            if (it == "201") {
-                Toast.makeText(activity, "User registration successful", Toast.LENGTH_LONG).show()
-            }
+        viewModel.messages.observe(this, Observer {
```

Comment:
This can be removed in favor of `viewModel.signupSuccessful`?

Reply:
Didn't get it. Can you please explain

Reply:
You are observing `viewModel.signupSuccessful`, the hardcoded message could be shown there, but no issue